org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/SonarLintCorePlugin.java

```diff
@@ -125,9 +126,18 @@ public void startupAsync() {
 
     @Override
     public IStatus run(IProgressMonitor monitor) {
-      telemetry.init();
+      startTelemetry();
       return Status.OK_STATUS;
     }
+
+    private void startTelemetry() {
+      if (SonarLintTelemetry.shouldBeActivated()) {
```

Comment:
I'm not totally sure of the value to move the condition here, compared to keeping everything related to telemetry in the SonarLintTelemetry class.

Reply:
It's a first step towards the design I suggested in another comment. My idea is to make explicit that we have 2 different situations, real telemetry in production and disabled telemetry for tests. Ideally I would represent that with 2 different implementations, and the choice between them should be made here IMO. I will try to implement that later this week